src/main/java/org/springframework/data/domain/Range.java

```diff
@@ -87,11 +206,230 @@ public boolean contains(T value) {
 
 		Assert.notNull(value, "Reference value must not be null!");
 
-		boolean greaterThanLowerBound = lowerBound == null ? true
-				: lowerInclusive ? lowerBound.compareTo(value) <= 0 : lowerBound.compareTo(value) < 0;
-		boolean lessThanUpperBound = upperBound == null ? true
-				: upperInclusive ? upperBound.compareTo(value) >= 0 : upperBound.compareTo(value) > 0;
+		boolean greaterThanLowerBound = lowerBound.getValue() //
+				.map(it -> lowerBound.isInclusive() ? it.compareTo(value) <= 0 : it.compareTo(value) < 0) //
+				.orElse(true);
+
+		boolean lessThanUpperBound = upperBound.getValue() //
+				.map(it -> upperBound.isInclusive() ? it.compareTo(value) >= 0 : it.compareTo(value) > 0) //
+				.orElse(true);
 
 		return greaterThanLowerBound && lessThanUpperBound;
 	}
+
+	/* 
+	 * (non-Javadoc)
+	 * @see java.lang.Object#toString()
+	 */
+	@Override
+	public String toString() {
+
+		String lower = this.lowerBound.getValue().map(Object::toString).map(it -> {
+
+			if (this.lowerBound.isInclusive()) {
```

Comment:
Ternary if? Could this be a package protected `getPrefix()`/`getSuffix()` method on `Bound`?